README.md

```diff
@@ -342,6 +350,57 @@ Also specify the principal corresponding to the client account identified by `oa
 
 This is not a full set of available `oauth.*` properties. All the `oauth.*` properties described in the next chapter about [configuring the Kafka clients](#configuring-the-kafka-client) also apply to configuring the client side of inter-broker communication. 
 
+#### Enabling the re-authentication
+
+Access tokens expire after some time. The token validation for the purpose of session authentication is only performed immediately after the new connection from the client is established. 
+If using SimpleACLAuthorizer or no authorizer at all, then there is no further need for the access token after the validation, and by default the expiry of the token will not result in session closure or denial of access within the existing session.
+
+Since Kafka version 2.2 the Kafka brokers support a [re-authentication mechanism](https://cwiki.apache.org/confluence/display/KAFKA/KIP-368%3A+Allow+SASL+Connections+to+Periodically+Re-Authenticate) allowing clients to update the token mid-session, without having to drop and re-establish the connection. 
+When a client sends the new access token, validation is performed on the broker as if a new connection was established.
+
+The mechanism has to be explicitly enabled on the Kafka broker by using the `connections.max.reauth.ms` property in `server.properties`:
+
+    connections.max.reauth.ms=3600000
+
+In this example the maximum time until next re-authentication is set to one hour.
+If the access token expires sooner than that, the re-authentication will be triggered sooner.
+
+The option can be specified per-listener. For example if you have a listener called `CLIENTS`, you can specify:
+
+    listener.name.clients.oauthbearer.connections.max.reauth.ms=3600000
+
+#### Enforcing the session timeout 
+
+If re-authentication is enabled, the session timeout is enforced as the expiry time of the access token. 
+By using re-authentication the multiple 'lightweight' sessions can follow one another over the same network connection for as long as the connection isn't closed or interrupted due to processes restarting or due to network issues. 
+
+If for some reason you can't enable re-authentication or don't want to use it, and if you want to invalidate the session when access token expires, but aren't using `KeycloakRBACAuthorizer`, which does this automatically (since version 0.6.0 of this library), you can use the `OAuthSessionAuthorizer` to enforce token expiry mid-session.
+
+`OAuthSessionAuthorizer` works by checking the access token expiry on every operation performed, and denies all access after the token has expired.
```

Comment:
Will we add support for this to Strimzi as well?

Reply:
This one no. I don't see the need. In order to not break compatibility with clients relying on current behaviour where session never expires due to re-authentication being disabled by default, I'll introduce the oauth authentication setting to turn on re-authentication. Once re-authentication is enabled, there is no need for `OAuthSessionAuthorizer`.